System.Containers.Tasks/ParseContainerProperties.cs

```diff
@@ -0,0 +1,94 @@
+using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.Linq;
+using System.Resources;
+using System.Text;
+using Microsoft.Build.Framework;
+using Microsoft.Build.Utilities;
+using System.Containers;
+
+#nullable disable
+
+namespace System.Containers.Tasks
+{
+    public class ParseContainerProperties : Microsoft.Build.Utilities.Task
+    {
+
+
+        /// <summary>
+        /// The full base image name. mcr.microsoft.com/dotnet/runtime:6.0, for example.
+        /// </summary>
+        [Required]
+        public string FullyQualifiedBaseImageName { get; set; }
+
+        /// <summary>
+        /// The image name for the container to be created.
+        /// </summary>
+        [Required]
+        public string ContainerImageName { get; set; }
+
+        /// <summary>
+        /// The tag for the container to be created.
+        /// </summary>
+        [Required]
+        public string ContainerImageTag { get; set; }
+
+        [Output]
+        public string ParsedContainerRegistry { get; private set; }
+
+        [Output]
+        public string ParsedContainerImage { get; private set; }
+
+        [Output]
+        public string ParsedContainerTag { get; private set; }
+
+        [Output]
+        public string NewContainerImageName { get; private set; }
+
+        [Output]
+        public string NewContainerTag { get; private set; }
+
+        public override bool Execute()
+        {
+            if (!ContainerHelpers.IsValidImageName(ContainerImageName))
+            {
+                Log.LogError("Invalid image name: {0}", ContainerImageName);
+                return !Log.HasLoggedErrors;
+            }
+
+            if (!ContainerHelpers.IsValidImageTag(ContainerImageTag))
+            {
+                Log.LogError("Invalid image tag: {0}", ContainerImageTag);
+                return !Log.HasLoggedErrors;
+            }
+
+            if (!ContainerHelpers.TryParseFullyQualifiedContainerName(FullyQualifiedBaseImageName,
+                                                                      out string outputReg,
+                                                                      out string outputImage,
+                                                                      out string outputTag))
+            {
+                Log.LogError("Could not parse the given ContainerBaseImage: {0}", FullyQualifiedBaseImageName);
+                return !Log.HasLoggedErrors;
+            }
+
+            ParsedContainerRegistry = outputReg;
+            ParsedContainerImage = outputImage;
+            ParsedContainerTag = outputTag;
+            NewContainerImageName = ContainerImageName;
+            NewContainerTag = ContainerImageTag;
+
+            if (BuildEngine != null)
+            {
+                Log.LogMessage("Parsed the following properties. Note: Spaces are replaced with dashes.");
```

Comment:
Why the space/dash thing?

Reply:
@baronfel suggested we do this

Reply:
container names can't have them, so we need some way to normalize arbitrary project names to an 'utterable' version. this is also common with e.g. cloud provider resource names.

Reply:
Note this would only apply to the 'fully qualified' string the user has to specify. If there's a space in the "new container image/tag," they'll fail the regex parse and be told as much.

Which makes me think it's worth it to either log a message or a warning if there are spaces in the user's 'fully qualified' image. A warning sounds more apprioriate